service/resource/cloudformation/current.go

```diff
@@ -1,7 +1,54 @@
 package cloudformation
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for AWS stack")
+
+	stackName := key.MainStackName(customObject)
+
+	describeInput := &awscloudformation.DescribeStacksInput{
+		StackName: aws.String(stackName),
+	}
+	describeOutput, err := r.awsClient.DescribeStacks(describeInput)
+
+	// FIXME: The validation error returned by the CloudFormation API doesn't make things easy to check, other than
+	// looking for the returned string. There's no constant in aws go sdk for defining this string, it comes from
+	// the service.
+	stackNotFoundError := fmt.Sprintf("Stack with id %s does not exist", stackName)
+	if strings.Contains(err.Error(), stackNotFoundError) {
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", notFoundError)
+		return StackState{}, nil
+
+	} else if err != nil {
+
+		return nil, microerror.Mask(err)
+
+	}
+
+	if len(describeOutput.Stacks) > 1 {
+		return nil, microerror.Mask(notFoundError)
+	}
+
+	outputStackState := StackState{
+		Name:    stackName,
+		Outputs: describeOutput.Stacks[0].Outputs,
+	}
+
+	return outputStackState, nil
```

Comment:
Above is `nil` returned. Now here is `StackState` returned. The types being returned should be aligned. Above the zero value would now be `StackType{}`. Or we decide to pass a pointer, then the code above is ok but we should return a pointer to the stack type here.

Reply:
Done, unit tests are missing here for uncovering this kind of things, i'll add them in a separate branch together with other methods that require mocking of the cf client.